contracts/metagov/VeBalHelper.sol

```diff
@@ -0,0 +1,112 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.0;
+
+import {VeBalDelegatorPCVDeposit} from "./VeBalDelegatorPCVDeposit.sol";
+import {Ownable} from "@openzeppelin/contracts/access/Ownable.sol";
+import {SafeERC20, IERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import {TransparentUpgradeableProxy, ProxyAdmin} from "@openzeppelin/contracts/proxy/transparent/ProxyAdmin.sol";
+
+/// @title Helper contract for veBAL OTC
+/// @author eswak
+contract VeBalHelper is Ownable {
+    using SafeERC20 for IERC20;
+
+    TransparentUpgradeableProxy public constant BOOST_RECIPIENT_PROXY =
+        TransparentUpgradeableProxy(payable(address(0x66977Ce30049CD0e443216Bf26377966c3A109E2)));
+
+    VeBalDelegatorPCVDeposit public immutable pcvDeposit;
+    IERC20 public immutable otcToken;
+    uint256 public immutable otcAmount;
+    address public immutable otcDestination;
+    address public immutable otcBuyer;
+
+    constructor(
+        address _owner,
+        address _pcvDeposit,
+        address _otcToken,
+        uint256 _otcAmount,
+        address _otcDestination,
+        address _otcBuyer
+    ) Ownable() {
+        _transferOwnership(_owner);
+        pcvDeposit = VeBalDelegatorPCVDeposit(_pcvDeposit);
+        otcToken = IERC20(_otcToken);
+        otcAmount = _otcAmount;
+        otcDestination = _otcDestination;
+        otcBuyer = _otcBuyer;
+    }
+
+    // ----------------------------------------------------------------------------------
+    // OTC Management
+    // ----------------------------------------------------------------------------------
+
+    /// @notice perform OTC
+    function otcBuy() external {
+        require(msg.sender == otcBuyer, "UNAUTHORIZED");
+        otcToken.safeTransferFrom(msg.sender, otcDestination, otcAmount);
+        _transferOwnership(msg.sender);
+        BOOST_RECIPIENT_PROXY.changeAdmin(msg.sender);
+    }
```

Comment:
Potentially need to update this if the OTC is performed in a less trustless way (Aave suggested holding DAI in a multisig separately from the ownership transfer)